### PR TITLE
add code of conduct

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR adds a `CODE_OF_CONDUCT.md` and `REPORT_HANDLING_MANUAL.md` as discussed in #302. The files added are almost identical to the CoC and manual used by both SciPy and NumPy - the only changes are to replace our project name `napari`, remove references to a higher steering committee as we do not have one at this point (though if we make one we can add those references back in), and then swapping out the names / emails of the individuals on the committee.

We still need to create a napari forwarding address that will forward on the the emails of the CoC committee. @kne42 said she had an idea on how to do this.

We also need to decide which of us will be on the CoC committee. SciPy currently has 3 people and I think a similar number is good for us. I'm happy to be on it and would also nominate @kne42 and @jni if they are happy to do it.  

Finally we should request one / two independent people that can be contact in case there are issues involving the committee. We can discuss this at our next team meeting, I have a couple ideas on who good people might be.

### CODE REVIEW

**Comment by sofroniewn** [comment]
@jni @kne42 @csweaver, I've now updated out CoC with our names and created a google group, `napari-conduct@googlegroups.com` to which I have invited @kne42 and @jni - only the three of us have access, and all content should be private. I used your personal emails, though can use other ones instead. We should double check that it is all working as expected.

@csweaver I've listed the email address you gave me as an optional 3rd party to contact. I'd like to add one more name there so it's not just you, but I think we can do that after this is merged.

The content of the CoC and reporting manual are almost identical to that from [SciPy](https://docs.scipy.org/doc/scipy/reference/dev/conduct/code_of_conduct.html) as we discussed before but with `napari` inserted instead and references to other steering committees dropped.

If @kne42, @jni, and @csweaver you can all look this over one more time and post your approval in this thread we can then merge.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+3. Be collaborative. Our work will be used by other people, and in turn we will depend on the work of others. When we make something for the benefit of the project, we are willing to explain to others how it works, so that they can build on the work to make it even better. Any decision we make will affect users and colleagues, and we take those consequences seriously when making decisions.
+
+4. Be inquisitive. Nobody knows everything! Asking questions early avoids many problems later, so we encourage questions, although we may direct them to the appropriate forum. We will try hard to be responsive and helpful.
+
+5. Be careful in the words that we choose. We are careful and respectful in our communication and we take responsibility for our own speech. Be kind to others. Do not insult or put down other participants. We will not accept harassment or other exclusionary behaviour, such as:
```
Comment (kne42) [inline CODE_OF_CONDUCT.md:21]: could you add something that has to do with misgendering participants who have explicitly requested to be addressed by specific pronouns

**Review by kne42 — APPROVED** [review]
lgtm – just one request

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (kne42) [reply CODE_OF_CONDUCT.md:21]: oh and something for ableist slurs

Comment (sofroniewn) [reply CODE_OF_CONDUCT.md:21]: sounds good to me. i've made the edits - let me know if they look good to you.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+> * Intentional or repeated misgendering of participants who have explicitly requested to be addressed by specific pronouns.
+> * Unwelcome sexual attention.
+> * Excessive profanity. Please avoid swearwords; people differ greatly in their sensitivity to swearing.
+> * Repeated harassment of others. In general, if someone asks you to stop, then stop.
+> * Advocating for, or encouraging, any of the above behaviour.
+
+## Diversity Statement
+The napari project welcomes and encourages participation by everyone. We are committed to being a community that everyone enjoys being part of. Although we may not always be able to accommodate each individual’s preferences, we try our best to treat everyone kindly.
+
+No matter how you identify yourself or how others perceive you: we welcome you. Though no list can hope to be comprehensive, we explicitly honour diversity in: age, culture, ethnicity, genotype, gender identity or expression, language, national origin, neurotype, phenotype, political beliefs, profession, race, religion, sexual orientation, socioeconomic status, subculture and technical ability, to the extent that these do not conflict with this code of conduct.
```
Comment (jni) [inline CODE_OF_CONDUCT.md:39]: "to the extent that these do not conflict with this code of conduct" Oooh, nice way to dodge the [paradox of tolerance](https://en.wikipedia.org/wiki/Paradox_of_tolerance). =D

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Probably fixing the failing build is important... =P

**Comment by sofroniewn** [comment]
I've updated this so our readme / contributing guidelines have links to the CoC. I had tried to find another person to be an external contact for our CoC, but no luck so far. I will merge as is though, as we have at least one external person.

**Pull request merged**